go/marshal/decode_test.go

```diff
@@ -750,3 +750,66 @@ func TestDecodeSetWrongMapType(t *testing.T) {
 	assert.Error(err)
 	assert.Equal(`Cannot unmarshal Map<Number, struct {}> into Go value of type map[int]struct {}, field has "set" tag`, err.Error())
 }
+
+func TestDecodeOmitEmpty(t *testing.T) {
+	assert := assert.New(t)
+
+	type S struct {
+		Foo int `noms:",omitempty"`
```

Comment:
I just wanted to ensure that nested structs were handled properly because that was my use case. I guess given the structure of the code there's no special reason these would fail, so I can remove this test if you prefer.